### PR TITLE
Update all python code using yapf

### PR DESCRIPTION
**Pull request by beets**
Also test for python lint errors, and add support for fixing that using run_test.py -f
Also reformatted yaml files using Prettier, though that is unenforced (cloudbuild.deploy.yaml already uses this style).

Might be easier to review by looking through each commit separately. Only the second commit should have logic changes, others were automatically done (first using yapf --recursive --i --style=google server/ tools/) and the latter using Prettier in vscode.

### CODE REVIEW

**Comment by beets** [comment]
@chejennifer @edumorlom fyi - this might cause some amount of merge conflict, but should be easily fixable (run that yapf command to fix your local change before doing a merge/rebase)

File: server/routes/api/stats.py [hunk 1]
```diff
@@ -20,12 +20,13 @@
 
 # Define blueprint
 bp = Blueprint(
```
Comment (shifucun) [inline server/routes/api/stats.py:22]: some how this one is not wrapped in one line..

File: run_test.sh [hunk 2]
```diff
@@ -23,23 +23,28 @@ function run_npm_test {
...
+# Tests if lint is required
+function run_lint_test {
   cd static
   npm list eslint || npm install eslint
   if ! npm run test-lint; then
     echo "\nFix lint errors by running ./run_test.sh -f"
     exit 1
   fi
   cd ..
+  if ! yapf --recursive --diff --style=google server/ tools/; then
```
Comment (shifucun) [inline run_test.sh:35]: need to install yapf first?

**Review by shifucun — DISMISSED** [review]
Do we want to use 2 spaces instead of 4 space across for python? 2 spaces seems used in Google..

File: run_test.sh (hunk 2 above)
Comment (beets) [reply run_test.sh:39]: updated the script to install if not found

File: server/routes/api/stats.py (hunk 1 above)
Comment (beets) [reply server/routes/api/stats.py]: Oh, that's odd. Manually made it one line

**Review by beets — COMMENTED** [review]
I think we should stick with 4, so it's just default yapf, and it's also what we are using in the data repo. I also find 4 spaces easier to read with python code.

**Comment by shifucun** [comment]
seeing some lint errors from the presubmit

**Comment by beets** [comment]
> Hm, those are just warnings, but the error is in installing yapf through
> pip. I'll try a few other things...


The virtual environment was not created successfully because ensurepip is
not
available. On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

apt-get install python3-venv

You may need to use sudo with that command. After installing the
python3-venv
package, recreate your virtual environment.

Failing command: ['/workspace/server/.env/bin/python3', '-Im', 'ensurepip',
'--upgrade', '--default-pip']

**Review by shifucun — APPROVED** [review]

**Comment by beets** [comment]
yay! finally got tests passing

**Pull request merged**